lua/package-info/modules/core/show.lua

```diff
@@ -111,5 +111,6 @@ return function()
             set_virtual_text(prod_dependencies, outdated_dependencies)
             set_virtual_text(peer_dependencies, outdated_dependencies)
         end)
+        config.highlight_groups(config.options.colors) -- XXX Weird hack
```

Comment:
Does it not work if the highlight is set before the virtual text?

In other words, the default highlight is set before the virtual text is set, so my question would be does it not work like that? 

Weird hack indeed :D

Reply:
I figured it out. My color scheme does something global and clears all highlights, wiping out the highlight command run  on setup(). I expect my color scheme is not the only one so ill mannered. One solution would be to require and document that package-info.setup() be run _after_ all color schemes or similar plugins. I think its more durable and less onerous on the plugin user to just set it in show() instead.

So what I've done is _only_ call register_highlight_groups() from show() now, removing the call from setup().
There is an argument to be made that this means I should move the register_highlight_group() function out of config.lau and into show.lua since it is only used there.

Reply:
Hmm, is this due to that specific color scheme or? It seems weird to me that it would just wipe everything on its own. 

And would this hack not be required if you loaded this plugin after the color scheme?

What I don't like about this approach is that highlight registration and setup should occur only once at startup. This way it's being forced to register each time which is not optional.

Reply:
I think I have a solution you'll like.
Any colorscheme resets all highlights no matter the order in the config files. I tried with several. Then I read this article: https://gist.github.com/romainl/379904f91fa40533175dfaec4c833f2f

So what I do with the latest commit is:
1. Register highlights on setup() only.
2. Create an autocmd that re-registers highlights only after the colorscheme command has finished.

I've removed the hacky highlight command in show()

I also exposed the register_highlight_groups() function at the package level in `package-info/init.lua`, but we could just as easily keep it concealed down in config.

Reply:
That makes more sense. I like it.

What would the purpose of exposing `register_highlight_groups` be?

Reply:
Really, because I am new to Lua and didn't understand how to call into config in the autocmd. But I just saw how to do that and removed register_highlight_groups from init.lua.

Now the only functional changes in this PR are to README.md & config.lua.
Also, there are some minor linting changes to init.lua and delete.lua other files that running stylua fixed.